cloudinstall/charms/__init__.py

```diff
@@ -253,7 +253,18 @@ def set_relations(self):
 
         Override in charm specific.
         """
-        if len(self.related) > 0:
+        if isinstance(self.related, dict):
+            for charm in self.related.keys():
+                try:
+                    log.debug("add_relation ({}, {})".format(
+                        *self.related[charm]))
+
+                    self.juju.add_relation(*self.related[charm])
```

Comment:
minor style comment here, I usually prefer explicitly unpacking tuples instead of using '*'. then you know what's actually being sent, and the code is a little clearer.